src/test/java/fi/aalto/cs/intellij/actions/EnablePluginsActionTest.java

```diff
@@ -0,0 +1,46 @@
+package fi.aalto.cs.intellij.actions;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.intellij.ide.plugins.IdeaPluginDescriptor;
+import com.intellij.ide.plugins.IdeaPluginDescriptorImpl;
+import com.intellij.notification.Notification;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.jdom.JDOMException;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class EnablePluginsActionTest {
+
+  @Test
+  public void testActionPerformed() throws IOException, JDOMException {
+    IdeaPluginDescriptorImpl ideaPluginDescriptor = getIdeaPluginDescriptor();
+    ideaPluginDescriptor.setEnabled(false);
+    List<IdeaPluginDescriptor> descriptors = new ArrayList<>();
+    descriptors.add(ideaPluginDescriptor);
+
+    assertFalse(ideaPluginDescriptor.isEnabled());
+
+    EnablePluginsAction enableMissingPluginsAction = new EnablePluginsAction("",
+        descriptors);
+    AnActionEvent anActionEvent = Mockito.mock(AnActionEvent.class);
+    Notification notification = Mockito.mock(Notification.class);
```

Comment:
**actions:** on the other hand, the logix performed within an action can always be wrapped into a function, and so there won't be a need to mock anything, you just test that function